app/src/main/res/values/strings.xml

```diff
@@ -328,4 +329,25 @@
     <string name="local_bundle_description">Import local files from your storage, does not automatically update</string>
     <string name="remote_bundle_description">Import remote files from a URL, can automatically update</string>
     <string name="recommended">Recommended</string>
+
+    <string name="installation_failed_dialog_title">Installation failed</string>
+    <string name="installation_cancelled_dialog_title">Installation cancelled</string>
+    <string name="installation_blocked_dialog_title">Installation blocked</string>
+    <string name="installation_conflict_dialog_title">Installation conflict</string>
+    <string name="installation_incompatible_dialog_title">Installation incompatible</string>
+    <string name="installation_invalid_dialog_title">Installation invalid</string>
+    <string name="installation_storage_issue_dialog_title">Installation storage issue</string>
+    <string name="installation_timeout_dialog_title">Installation timeout</string>
+    <string name="installation_success_dialog_title">Installation succeeded</string>
+    <string name="installation_failed_description">The installation failed due to an unknown reason. Try again?</string>
+    <string name="installation_aborted_description">The installation was cancelled manually. Try again?</string>
+    <string name="installation_blocked_description">The installation was blocked. Review your device security settings and try again.</string>
+    <string name="installation_conflict_description">The installation was prevented by an existing installation of the app. Uninstall the installed app and try again?</string>
+    <string name="installation_incompatible_description">The app is incompatible with this device. Contact the developer of the app and ask for support.</string>
```

Comment:
Ideally, this should *probably* be checked before patching in the first place, you can place a dismissable warning dialog or something. The string looks fine though.

Reply:
This PR is primarily introducing a Dialog that handles status codes and accepts a view model for taking action for certain status codes Introducing such a warning is out of scope for the PR.